src/routes/conversation/[id]/web-search/+server.ts

```diff
@@ -73,45 +63,63 @@ export async function GET({ params, locals, url }) {
 
 				appendUpdate("Searching Google", [webSearch.searchQuery]);
 				const results = await searchWeb(webSearch.searchQuery);
-
-				let text = "";
 				webSearch.results =
 					(results.organic_results &&
-						results.organic_results.map((el: { link: string }) => el.link)) ??
+						results.organic_results.map((el: { title: string; link: string }) => {
+							const { title, link } = el;
+							const { hostname } = new URL(link);
+							return { title, link, hostname };
+						})) ??
 					[];
-
-				if (results.answer_box) {
-					// if google returns an answer box, we use it
-					webSearch.answerBox = JSON.stringify(removeLinks(results.answer_box));
-					text = webSearch.answerBox;
-					appendUpdate("Found a Google answer box");
-				} else if (results.knowledge_graph) {
-					// if google returns a knowledge graph, we use it
-					webSearch.knowledgeGraph = JSON.stringify(removeLinks(results.knowledge_graph));
-					text = webSearch.knowledgeGraph;
-					appendUpdate("Found a Google knowledge page");
-				} else if (webSearch.results.length > 0) {
-					let tries = 0;
-
-					while (!text && tries < 3) {
-						const searchUrl = webSearch.results[tries];
-						appendUpdate("Browsing result", [JSON.stringify(searchUrl)]);
+				webSearch.results = webSearch.results
+					.filter(({ link }) => !link.includes("youtube.com")) // filter out youtube links
```

Comment:
would love to know if there's a banlist somewhere for websites we know we can't easily parse